### ISSUE TITLE
Disable AP

### ISSUE THREAD

**Issue by jay-oswald** [meatpiHQ/wican-fw#111]
I would like to have the option to disable AP on the WiCan

The way my other smart devices work is I can disable AP, but if the device can't connect to my home WiFi AP is re enabled temporarily to enable it to he reconfigured.

There is also a limit with Bluetooth where it can only be on when it's in AP only mode, I'm also hoping that it could work with station plus Bluetooth. As I would like to be able to use home wifi at home, then connect via Bluetooth to use car scanner app when away from home, which isn't possible at the moment.

**Comment by njh** [comment]
Yes please!

I would rather not go round advertising that I have a WiCAN device in my car.

**Comment by meatpiHQ** [comment]
Ok looks like this is the most popular feature request, I'll add this in the next update. But the problem is that when you're on the road away from your wifi it will still be broadcasting. I did solve this by adding a button on WiCAN Pro but unfortunately WiCAN doesn't have a button.

**Comment by jay-oswald** [comment]
@meatpiHQ could this logic work?

Mode 1: AP only, no station (default)

Mode 2:

Leave station mode on constantly.

When device is first plugged in enable AP for 5 mins. If nothing connects in those 5 mins it just turns off.

That way if someone needs to update details, they can unplug it, then plug it back in to enable the AP and connect to it. Could also then add an option on web UI to reboot without AP to disable again after your done.

Would let the AP be off 99% of the time, and easy enough to enable it by unplugging and replugging when needed

**Comment by njh** [comment]
@jay-oswald I like that idea. Is it standard that OBD-II ports are always powered, even if ignition is off?

**Comment by jay-oswald** [comment]
@njh yes there are constant power pins on obd2 port. Which im pretty sure wican already uses

**Comment by stav242** [comment]
yes please. just want the AP to stop broadcasing, i only use station mode. i dont need the AP - its a exposed vulnerability being there if not required.

**Comment by T1983M** [comment]
> [@meatpiHQ](https://github.com/meatpiHQ) could this logic work?
> 
> Mode 1: AP only, no station (default)
> 
> Mode 2:
> 
> Leave station mode on constantly.
> 
> When device is first plugged in enable AP for 5 mins. If nothing connects in those 5 mins it just turns off.
> 
> That way if someone needs to update details, they can unplug it, then plug it back in to enable the AP and connect to it. Could also then add an option on web UI to reboot without AP to disable again after your done.
> 
> Would let the AP be off 99% of the time, and easy enough to enable it by unplugging and replugging when needed

This logic could also be applied to the Wakeup 12V High and 12V Low Sleep current with the delay. Currently the delay is 1 minute as minimum, but this causes trouble when locking modern cars and the CAN is still queried. 
If you plug the Dongle in it stays awake for 5 min regardless of the settings. This would allow to correct any false set values, and in usage do not set of the alarm when locking the car in the car park to quickly.

**Comment by NilsRo** [comment]
> Yes please!
> 
> 
> 
> I would rather not go round advertising that I have a WiCAN device in my car.
> 
> 

Me too. It's a unnecessary security risk as for Home Assistant Cases AP is not necessary. Would be nice to have it as a backup after powering on for some minutes only as without a ESP IDF issue could used (if there is a unclosed bug). Coding should be easy and not memory consuming.

**Comment by tweak19** [comment]
Any Updates?

**Comment by meatpiHQ** [comment]
@tweak19 already implemented.

**Comment by tweak19** [comment]
In what kind of firmware? I still see the access point if the wican isn't connected to the local WiFi and working as station/client. 
Thx!

**Comment by stav242** [comment]
> In what kind of firmware? I still see the access point if the wican isn't connected to the local WiFi and working as station/client. Thx!

this one probably ...
https://github.com/meatpiHQ/wican-fw/pull/496

**Comment by tweak19** [comment]
Hi, 
I tried it today again, but it's not working for me. Or do I expect something what's not available?

I connected the wican and the wican is working as station, and as already discussed, I expect, that the AP is disabled after few minutes. 

But the ap is available, still after 30 min.

Kind regards

**Comment by hurzhurz** [comment]
Has anyone tried my suggested change https://github.com/meatpiHQ/wican-fw/pull/496 yet?
There is a build with this (and some other) changes avaliable here: https://github.com/hurzhurz/wican-fw/releases/tag/v412-103-gccf2a32
Bute be careful, depending on from which version you are coming, it might mess up your settings und you may need to re-flash using USB: https://meatpihq.github.io/wican-fw/config/firmware-update#_2-usb-flash-for-wican

**Comment by loryanstrant** [comment]
This is working for me

**Comment by tweak19** [comment]
+1 works also for me. 
Only poor bug in 4.12 is the vehicle import. 

@hurzhurz seems there is an issue with the sleep mode. I have the problem that I have a huge battery drain with this fw.
I will try to compare if this even happens with the original fw.
